### PR TITLE
add item get by area

### PR DESCRIPTION
**Pull request by festinalente91**
- added get items by area
  -  you can now query items through lat, lng, zoom
  - checkout apidoc GET items/ for the details
- added new initMock function
  - use it for your test, just call initMock() before test
    @TODO @ovekyc @HewonJeong  need to update get all items later

### CODE REVIEW

**Comment by hewonjeong** [comment]
LGTM

File: src/server/database-mock-data.js [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+  'image-0-item-8523910540002-b82e-473b-1234-ead0f190b003-8523570664000-b82e-473b-1234-ead0fts0b000': {
+    key: 'image-8523570664000-b82e-473b-1234-ead0fts0b000'
+  },
+  'image-0-item-8523910540002-b82e-473b-1234-ead0f190b003-8523571664000-b82e-473b-1234-ead0fts43200': {
+    key: 'image-8523571664000-b82e-473b-1234-ead0fts43200'
+  },
+  'image-0-item-8523910540002-b82e-473b-1234-ead0f190b003-8523574664000-b82e-473b-1234-ead0f54gvr00': {
+    key: 'image-8523574664000-b82e-473b-1234-ead0f54gvr00'
+  }
+}
```
Comment (hewonjeong) [inline src/server/database-mock-data.js:483]: Thank you for your hard work!

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -7,37 +7,64 @@ import {S3Connector} from '../aws-s3';
...
           // @TODO we have to limit the number of items.
+          const imagePromises = [];
           db.createReadStream({
-            start: `item-${STATE.ALIVE}-${key}-`,
-            end: `item-${STATE.ALIVE}-${key}-\xFF`
+            start: `${ENTITY.ITEM}-${STATE.ALIVE}-${key}-`,
+            end: `${ENTITY.ITEM}-${STATE.ALIVE}-${key}-\xFF`
           }).on('data', (data) => {
-            db.get(data.value.ref, (err, refData) => {
+            db.get(data.value.key, (err, refData) => {
```
Comment (HewonJeong) [inline src/server/items/controllers.js]: How bout idxData?

File: src/server/items/routes.js [hunk 3]
```diff
@@ -15,33 +15,45 @@ const router = express.Router();
...
+ *         {
+ *            "title": "Union Square Public Toilet",
+ *            "lat": 37.7632684,
+ *            "lng": -122.4182374,
+ *            "address": "2295 Harrison St, San Francisco, CA 94110, United States",
+ *            "createdDate": "2016-10-08T01:11:46.851Z",
+ *            "modifiedDate": "2016-10-08T01:11:46.851Z",
+ *            "category": "facility",
+ *            "startTime": "2016-10-08T01:11:46.851Z",
+ *            "endTime": "2016-10-20T01:11:46.851Z",
```
Comment (hewonjeong) [inline src/server/items/routes.js:28]: miss userKey field
"userKey": "user-8523574664000-b82e-473b-1234-ead0f54gvr00",

File: src/server/items/routes.js [hunk 4]
```diff
@@ -15,33 +15,45 @@ const router = express.Router();
...
- *     }
+ *            "title": "Cafe Free Wifi",
+ *            "lat": 37.7652022,
+ *            "lng": -122.4201257,
+ *            "address": "2500 17th St, San Francisco, CA 94110, United State",
+ *            "createdDate": "2016-10-09T01:11:46.851Z",
+ *            "modifiedDate": "2016-10-09T01:11:46.851Z",
+ *            "category": "facility",
+ *            "startTime": "2016-10-09T01:11:46.851Z",
+ *            "endTime": "2016-10-19T01:11:46.851Z",
```
Comment (hewonjeong) [inline src/server/items/routes.js:46]: ditto

File: src/server/items/controllers.js [hunk 5]
```diff
@@ -7,37 +7,64 @@ import {S3Connector} from '../aws-s3';
...
       for (const key of keys) {
         promises.push(new Promise((resolve, reject) => {
           // @TODO we have to limit the number of items.
+          const imagePromises = [];
           db.createReadStream({
-            start: `item-${STATE.ALIVE}-${key}-`,
-            end: `item-${STATE.ALIVE}-${key}-\xFF`
+            start: `${ENTITY.ITEM}-${STATE.ALIVE}-${key}-`,
+            end: `${ENTITY.ITEM}-${STATE.ALIVE}-${key}-\xFF`
           }).on('data', (data) => {
```
Comment (hewonjeong) [inline src/server/items/controllers.js:24]: How bout idxData?

File: src/server/items/controllers.js [hunk 6]
```diff
@@ -7,37 +7,63 @@ import {S3Connector} from '../aws-s3';
...
-        }));
+        }))
       }
-      Promise.all(promises).then(() => {
-        res.status(200).send({
-          items
-        });
-        cb();
+      Promise.all(promises).then(()=>{
+          res.status(200).send({
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Why

Comment (festinalente91) [reply src/server/items/controllers.js]: fixed

File: src/server/items/routes.js (hunk 3 above)
Comment (festinalente91) [reply src/server/items/routes.js:28]: solved

File: src/server/items/routes.js (hunk 4 above)
Comment (festinalente91) [reply src/server/items/routes.js:46]: solved

**Comment by ovekyc** [comment]
LGTM

**Comment by festinalente91** [comment]
give me lgtm

**Pull request merged**